# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/startup/nsAppStartup.cpp
# Commit: 765d3364cca0
# Full Hash: 765d3364cca0a67c614cb4fb5dff413a8c29d834
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2020-04-07 03:53:09
# Regressor Bug: 1616353
# File Overlap Count: 1
# Description:
#   Bug 1616353 - Part 7.2: Create and use nsOpenWindowInfo types in nsWindowWatcher logic, r=kmag
#   
#   This patch builds on top of part 7.1 by creating this object within
#   nsWindowWatcher and ContentParent to carry the relevant information through
#   provider interfaces when creating new content windows. The nsOpenWindowInfo
# ==============================================================================

diff -r 8a13355b4ac4 -r 765d3364cca0 toolkit/components/startup/nsAppStartup.cpp
--- a/toolkit/components/startup/nsAppStartup.cpp	Mon Apr 06 14:29:59 2020 +0000
+++ b/toolkit/components/startup/nsAppStartup.cpp	Mon Apr 06 14:30:02 2020 +0000
@@ -582,10 +582,8 @@
 NS_IMETHODIMP
 nsAppStartup::CreateChromeWindow(nsIWebBrowserChrome* aParent,
                                  uint32_t aChromeFlags,
-                                 nsIRemoteTab* aOpeningTab,
-                                 mozIDOMWindowProxy* aOpener,
-                                 uint64_t aNextRemoteTabId, bool* aCancel,
-                                 nsIWebBrowserChrome** _retval) {
+                                 nsIOpenWindowInfo* aOpenWindowInfo,
+                                 bool* aCancel, nsIWebBrowserChrome** _retval) {
   NS_ENSURE_ARG_POINTER(aCancel);
   NS_ENSURE_ARG_POINTER(_retval);
   *aCancel = false;
@@ -612,14 +610,14 @@
                  "may work.");
 
     if (appParent)
-      appParent->CreateNewWindow(aChromeFlags, aOpeningTab, aOpener,
-                                 aNextRemoteTabId, getter_AddRefs(newWindow));
+      appParent->CreateNewWindow(aChromeFlags, aOpenWindowInfo,
+                                 getter_AddRefs(newWindow));
     // And if it fails, don't try again without a parent. It could fail
     // intentionally (bug 115969).
   } else {  // try using basic methods:
-    MOZ_RELEASE_ASSERT(aNextRemoteTabId == 0,
-                       "Unexpected aNextRemoteTabId, we shouldn't ever have a "
-                       "next actor ID without a parent");
+    MOZ_RELEASE_ASSERT(!aOpenWindowInfo,
+                       "Unexpected aOpenWindowInfo, we shouldn't ever have an "
+                       "nsIOpenWindowInfo without a parent");
 
     /* You really shouldn't be making dependent windows without a parent.
       But unparented modal (and therefore dependent) windows happen
@@ -633,8 +631,7 @@
 
     appShell->CreateTopLevelWindow(
         0, 0, aChromeFlags, nsIAppShellService::SIZE_TO_CONTENT,
-        nsIAppShellService::SIZE_TO_CONTENT, aOpeningTab, aOpener,
-        getter_AddRefs(newWindow));
+        nsIAppShellService::SIZE_TO_CONTENT, getter_AddRefs(newWindow));
   }
 
   // if anybody gave us anything to work with, use it